packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -249,7 +249,7 @@ BuildApp() {
   fi
 
   StreamOutput " ├─Assembling Flutter resources..."
-  RunCommand "${FLUTTER_ROOT}/bin/flutter" --suppress-analytics             \
+  RunCommand "${FLUTTER_ROOT}/bin/flutter"                                  \
```

Comment:
It would be nice to avoid double logging re-entrant calls back into Flutter such as run->xcode_backend.sh->bundle.

Could we set a flag so that only the outermost invoked Flutter CLI command gets analytics and all nested calls don't?

Reply:
Done.

Reply:
I think we need a more nuanced solution, here, though not needed for this CL. For example it would be nice to have timing events from re-entrant calls for build steps etc.

Reply:
if we want screen/event/timing time breakdown, we likely want to consider something more nuanced than GA. Otherwise, non session joined nested acts double logged as siblings may not be as actionable as data.

Reply:
1. We can pass whatever information we'd like to pass to the re-entrant calls.
2. Logging behavior can be different in a re-entrant context.
We will probably need to do both of those things whether GA is the backend or something else.

Reply:
+1

Reply:
What I meant by sibling log entries is that

- Yes, we can tag reentrant flutter build bundle to say that their gradle/shell script parent caller's parent was either a flutter command or not.
- But we won't be able to answer the questions of "for all the calls to flutter run that was 30s+, how much time did flutter build bundle take in median" without a self definable schema that allows sessionization beyond what the ga_sessions_* table lets us do. 
- We can repurpose custom dimensions of each top level command to bundle sub-command timing breakdowns as strings but then we're re-inventing the wheel.

In other words, whether GA is the backend or something implies pre-defined logging schema or not.

Reply:
I'm sure that's all true, but I don't think we should predicate collecting data from re-entrant flutter_tool invocations on switching to a new backend. For example, if ~all bundle subcommand invocations come from re-entrant invocations, then we can't make any queries about that information because we don't have it, let alone complex queries.